pkg/controllers/migration/virtualmachine.go

```diff
@@ -612,5 +641,13 @@ func (h *virtualMachineHandler) checkAndCreateVirtualMachineImage(vm *migration.
 			SourceType:  "download",
 		},
 	}
+
+	if vm.Spec.StorageClass != "" {
+		// update storage class annotations
+		vmi.Annotations = map[string]string{
+			"harvesterhci.io/storageClassName": vm.Spec.StorageClass,
+		}
+	}
+
```

Comment:
Just curious, Is it only for debug usage?

Reply:
harvester ui defines this annotation and this is to only keep it consistent if we introduce some change in the future that may be depending on this.